.github/workflows/clp-artifact-build.yaml

```diff
@@ -28,7 +28,7 @@ concurrency:
 jobs:
   filter-relevant-changes:
     name: "filter-relevant-changes"
-    runs-on: "ubuntu-24.04"
+    runs-on: ["self-hosted", "X64", "ubuntu-noble"]
```

Comment:
is it possible to update some docs to explain how those labels are created (i.e., what machines are backing them), and where can we find the lists of all labels?

maybe https://docs.yscope.com/clp/main/dev-docs/tooling-gh-workflows.html is a good place to add such info?